File: Utt2Database/PlayerStat.vb
Imports System
Imports System.Collections.Generic
Imports Microsoft.VisualBasic

Namespace Utt2Database
    Partial Public Class PlayerStat
        Public Property Id As Long

        Public Property PlayerId As Long

        Public Property ServerId As Long

        Public Property GameTime As Integer

        Public Property Score As Long

        Public Property Deaths As Integer

        Public Property LastMatchId As Long

        Public Overridable Property LastMatch As ServerMatch

        Public Overridable Property Player As Player

        Public Overridable Property Server As Server
    End Class
End Namespace

File: src/IndexerSharp/Utt2Database/PlayerStat.cs
namespace Naomai.UTT.Indexer.Utt2Database;
public class PlayerStat
{
    public long? Id { get; set; }
    public long PlayerId { get; set; }
    public long ServerId { get; set; }
    public int GameTime { get; set; }
    public long Score { get; set; }
    public int Deaths { get; set; }
    public long LastMatchId { get; set; }
    public virtual ServerMatch LastMatch { get; set; } = null!;
    public virtual Player Player { get; set; } = null!;
    public virtual Server Server { get; set; } = null!;

    public override string ToString()
    {
        return "PS" + PlayerId + "@S" + ServerId;
    }
}